src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProjectOptionsProcessor.cs

```diff
@@ -70,23 +64,12 @@ private bool ReparseCommandLineIfChanged_NoLock(ImmutableArray<string> arguments
 
         _commandLineChecksum = checksum;
 
-        // Dispose the existing stored command-line and then persist the new one so we can
-        // recover it later.  Only bother persisting things if we have a non-empty string.
-
-        _commandLineStorageHandle = null;
-        if (!arguments.IsEmpty)
-        {
-            using var stream = SerializableBytes.CreateWritableStream();
-            using var writer = new StreamWriter(stream);
-
-            foreach (var value in arguments)
-                writer.WriteLine(value);
+        ReparseCommandLine_NoLock(arguments);
 
-            writer.Flush();
-            _commandLineStorageHandle = _temporaryStorageService.WriteToTemporaryStorage(stream, CancellationToken.None);
-        }
+        // Only bother storing the command line if there is an effective ruleset, as that may
+        // require a later reparse using it.
+        _commandLine = GetEffectiveRulesetFilePath() != null ? arguments : [];
```

Comment:
```suggestion
        _commandLine = GetEffectiveRulesetFilePath() != null ? arguments : default;
```

Because then you can assert in ReparseCommandLine_NoLock that it shouldn't ever be default. If it is, then that means we screwed up somewhere and lost the command line when we shouldn't have, which would break the IDE experience pretty badly.

Reply:
Good idea